### ISSUE TITLE
number knob ignores comma as decimal separator

### ISSUE THREAD

**Issue by sgonzalez-orangeloops** [storybookjs/react-native#132]
**Describe the bug**
The number knob ignores numbers written after a comma. This is not desired when the device is configured with certain regions, like Russia, because in those cases iOS will pop up a keyboard with no option to write a dot.

**To Reproduce**
1. Set the region of a iOS device to russia (no need to change the language).
2. Add a number knob to a story.
3. Try to write number with decimals in the knob input.

**Expected behavior**
The number knob text input recognizes both the comma and the dot as decimal separators.

**Screenshots**
![ezgif-3-f2bbea0cc193](https://user-images.githubusercontent.com/56684338/99079419-ae0bc700-259e-11eb-97fa-76b41539dc4c.gif)

**System:**
```
System:
  OS: macOS 10.15.7
  CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
Binaries:
  Node: 15.0.1 - /usr/local/bin/node
  npm: 6.14.8 - /usr/local/bin/npm
Browsers:
  Chrome: 86.0.4240.193
  Edge: 86.0.622.63
  Safari: 14.0
npmPackages:
  @storybook/addon-actions: 5.3.18 => 5.3.18 
  @storybook/addon-knobs: 5.3.18 => 5.3.18 
  @storybook/addon-ondevice-actions: 5.3.18 => 5.3.18 
  @storybook/addon-ondevice-knobs: 5.3.23 => 5.3.23 
  @storybook/addons: 5.3.18 => 5.3.18 
  @storybook/react-native: 5.3.23 => 5.3.23 
  @storybook/react-native-server: 5.3.18 => 5.3.18
```

**Comment by sgonzalez-orangeloops** [comment]
Found [this commit](https://github.com/storybookjs/react-native/commit/1e787fa5bcd2e4504cdf7a0cb8f0b14382062558) that removes logic that might be helpful in this case, but I couldn't find context for that removal.

**Comment by dannyhw** [comment]
Was released as part of ondevice-knobs v5.3.24
